lib/gemfury/command/app.rb

```diff
@@ -295,6 +315,65 @@ def push_files(command, gem_paths)
     end
   end
 
+  C50K = 50000
+
+  class UploadIO
+    attr_reader :content_type, :original_filename, :local_path, :io
+
+    def initialize(filename_or_io, content_type = 'application/octet-stream', fname = nil)
+      io = filename_or_io
+      local_path = ''
+
+      if io.respond_to? :read
+        local_path = filename_or_io.respond_to?(:path) ? filename_or_io.path : 'local.path'
+      else
+        io = File.open(filename_or_io)
+        local_path = filename_or_io
+      end
+
+      fname ||= local_path
+
+      @content_type = content_type
+      @original_filename = File.basename(fname)
+      @local_path = local_path
+      @io = io
+
+      if @io.respond_to? :size
+        filesize = @io.size
+      else
+        filesize = @io.stat.size
+      end
+
+      if filesize > C50K
+        title = 'Uploading %s ' % File.basename(fname)
+        @bar = ProgressBar.create(:title => title, :total => filesize)
+      else
+        @bar = nil
+      end
+    end
+
+    def show_bar?
+      @bar != nil
+    end
+
+    def read(length)
+      buf = @io.read(length)
+      unless @bar.nil? || buf.nil?
+        @bar.progress += buf.bytesize
+      end
+
+      buf
+    end
+
+    def method_missing(*args)
+      @io.send(*args)
+    end
+
+    def respond_to?(meth, include_all = false)
```

Comment:
Can you do `UploadIO` this as a `SimpleDelegator` to handle `respond_to?` and `method_missing`? Also, maybe `ProgressIO` better reflects the name?